src/webgpu/api/validation/encoding/encoder_open_state.spec.ts

```diff
@@ -0,0 +1,383 @@
+export const description = `
+Validation tests to all commands of GPUCommandEncoder, GPUComputePassEncoder, and
+GPURenderPassEncoder when the encoder is not finished.
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../common/util/data_tables.js';
+import { unreachable } from '../../../../common/util/util.js';
+import { ValidationTest } from '../validation_test.js';
+
+import { beginRenderPassWithQuerySet } from './queries/common.js';
+
+class F extends ValidationTest {
+  beginRenderPass(commandEncoder: GPUCommandEncoder): GPURenderPassEncoder {
+    const attachmentTexture = this.device.createTexture({
+      format: 'rgba8unorm',
+      size: { width: 16, height: 16, depthOrArrayLayers: 1 },
+      usage: GPUTextureUsage.RENDER_ATTACHMENT,
+    });
+    this.trackForCleanup(attachmentTexture);
+    return commandEncoder.beginRenderPass({
+      colorAttachments: [
+        {
+          view: attachmentTexture.createView(),
+          clearValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+          loadOp: 'clear',
+          storeOp: 'store',
+        },
+      ],
+    });
+  }
+}
+
+export const g = makeTestGroup(F);
+
+type EncoderCommands = keyof Omit<
+  GPUCommandEncoder,
+  '__brand' | 'label' | 'finish' | 'beginComputePass' | 'beginRenderPass'
```

Comment:
I think the validation in beginComputePass/beginRenderPass should be pretty similar, maybe they shouldn't be omitted here. We'll probably have other tests that cover it, but if it's trivial to test them here too, then why not. (Can just make this a TODO for now. Better to work on the other comments first)

Reply:
Ok, I update the test to test begindComputePass/beginRenderPass.